src/main/java/com/example/ShoppingBasket.java

```diff
@@ -0,0 +1,53 @@
+package com.example;
+
+/**
+ * A shopping basket is a thing in which customer puts goods in order to buy it.
+ *
+ * <p>The customer can choose a basket suitable for his purchases.
+ *
+ * <p>The absence of a shopping basket does not prevent the customer from purchasing the goods.
+ *
+ * <p>The customer can change his basket or take another one.
+ */
+abstract class ShoppingBasket implements ShoppingCart {
+    /**
+     * The customer uses shopping basket.
+     */
+    public Customer customer;
+
+    /**
+     * The shopping basket holds up to 15 items with a total weight of up to 5 kg.
+     */
+    Goods[] itemsInBasket = new Goods[15];
+
+    /**
+     * Before purchasing customer should choose a suitable for his purchases shopping basket.
+     */
+    public void choose() {
```

Comment:
Choose what? Suppose I choose a product, what happens then?  How does the method signature tell about this?